main.go

```diff
@@ -65,6 +66,22 @@ func main() {
 		os.Exit(1)
 	}
 
+	// KUBECONFIG is required by the Helm library, and other tools that might be used by the adapter in the future.
+	kubeconfig := path.Join(
+		config.KubeConfigDefaults[configprovider.FilePath],
+		fmt.Sprintf("%s.%s", config.KubeConfigDefaults[configprovider.FileName], config.KubeConfigDefaults[configprovider.FileType]))
+	envval, present := os.LookupEnv("KUBECONFIG")
```

Comment:
@mgfeller I am not sure if we should be looking for `KUBECONFIG` in the adapter's environment because the kubeconfig path that we are setting is the one that adapter receives from the meshery server. Disregarding the `KUBECONFIG` from the environment will maybe ensure that the adapter communicates (if it does) with the same kubernetes cluster as the meshery server.

I maybe missing a use case here though :sweat_smile:

Reply:
@utkarsh-pro the use case is that developing, debugging, and testing happens in the local environment, i.e. running in-process, e.g. on the developers notebook. Just replacing `KUBECONFIG` messes up the local environment. So this is to be defensive and avoid time spending fixing and troubleshooting this (as I had to, unfortunately). Maybe there is another use case as well...

The meshery server sends the content of the relevant kube-config, not a file path, which the adapter saves in the file given by `kubeconfig` variable. The meshery server (if it is used at all in this use case) uses that same config content (not sure whether and where it is saved). 

Maybe that file path should be added at the start of `KUBECONFIG`?

If you have another robust solution I'm all ears :smile:

Reply:
I don't think golang would be able to change the environmental variable of the parent process as it internally uses the default C environment to set and get these variables, nothing fancier. (https://github.com/golang/go/blob/master/src/runtime/env_posix.go#L53-L59)

But yeah, I think this would be helpful for the debugging purposes. I just forgot to consider this scenario. Thanks for elaborating! :smile:

Reply:
@utkarsh-pro thanks for elaborating about handling environment variables. I wondered about that when I wrote the previous comment, I'm sure there was something that didn't work when I just replaced `KUBECONFIG` but I can't remember the details. Should've written it down. I'll test again if I find the time.

If I understand your comment correctly, golang gets the env var from the parent process, but the change is not propagated back.

So, just replacing the env var, instead of adding the new path to it, should be completely safe?

Reply:
I'll test that again and remove updating the env var, just setting it, if it works.

Reply:
@utkarsh-pro looks better now, and the BATS tests are successful also for Helm install.

Reply:
@mgfeller wow I am glad it worked! :smile: